app/assets/javascripts/discourse/controllers/group.js.es6

```diff
@@ -34,11 +41,13 @@ export default Ember.Controller.extend({
     });
   },
 
-  tabs: [
-    Tab.create({ name: 'members', active: true, 'location': 'group.index' }),
-    Tab.create({ name: 'posts' }),
-    Tab.create({ name: 'topics' }),
-    Tab.create({ name: 'mentions' }),
-    Tab.create({ name: 'messages' }),
-  ]
+  @computed('model.is_member')
+  getTabs(isMember) {
+    if (isMember) {
+      if (this.get('tabs').length === 4) this.get('tabs').push(Tab.create({ name: 'messages' }));
```

Comment:
I don't think checking the length of the tabs array and mutating it here is the safest approach. If we change the tabs these numbers would need to be updated and I think it would be easy for a developer to miss it.

Instead why not something like this:

```javascript
return this.get('tabs').filter(t => isMember || t.get('name') !== 'members');
```

Or even better, add `requiresMembership: true` to the `Tab.create` for members, then:

```javascript
return this.get('tabs').filter(t => isMember || !t.get('requiresMembership'));
```

That way we don't have to hard code the name.